src/store/index.ts

```diff
@@ -0,0 +1,101 @@
+import { createDirectStore, ModulesImpl } from 'direct-vuex';
+import getUrlParameter from '@/helpers/get-url-parameter';
+import { IMessage } from '@/types/c-notification';
+import session from './modules/session';
+import notification from './modules/notification';
+import breadcrumb from './modules/breadcrumb';
+
+interface IModules {
+  session: ModulesImpl;
+  breadcrumb: ModulesImpl;
+  notification: ModulesImpl;
+}
+
+export interface IStore {
+  modules: IModules;
+}
+
+interface IAction {
+  data: object;
+  messages: {
+    INFO: [];
+    ERROR: [];
+    SUCCESS: [];
+  };
+  assetsPath: string;
+}
+
+interface IInitialData {
+  [key: string]: IAction;
+}
+
+declare global {
+  interface Window { initialData: IInitialData; }
+}
+
+const data: IInitialData = window.initialData || {};
+let initialDataMessages: IMessage[] = [];
+
+const {
+  store,
+  rootActionContext,
+  moduleActionContext,
+  rootGetterContext,
+  moduleGetterContext
+} = createDirectStore({
+  modules: {
+    session,
+    notification,
+    breadcrumb,
+  }
+});
+
+// Set initial data
+Object.keys(data).forEach((action) => {
+  const { messages } = data[action] || {};
+
+  const storeModule = action.split('/')[0] as keyof IModules;
```

Comment:
Please only execute `split()` once. You could use a deconstruction:

```
const [storeModule, moduleAction] = action.split('/');
```


But also note: modules might be nested (e.g. `user/favorites/data`). If this is not possible with the new wrapper we need to inform the developers.

Reply:
⚠️ as already mentioned I would recommend we use the traditional way to dispatch the action here, because it seems a lot simpler to handle nested modules.

Reply:
replaced with the old logic